job_templates/regenerate_puppet_certificate.erb

```diff
@@ -0,0 +1,51 @@
+<%#
+name: Regenerate puppet certificate
+job_category: Miscellaneous
+snippet: false
+template_inputs:
+- name: Delete certificate on Puppet CA
+  required: true
+  input_type: user
+  description: If yes is selected, Foreman will contact Puppet CA of a host and remove
+    the certificate from there. However this requires safe mode to be disabled.
+  options: "no\r\nyes"
+  advanced: false
+provider_type: SSH
+kind: job_template
+%>
+
+<%#
+name: Regenerate puppet certificate
+job_category: Puppet
+description_format: Regenerate puppet certificate, optionally (with safe mode disabled, it can also remove the certificate from puppet ca)
+snippet: false
+provider_type: SSH
+kind: job_template
+%>
+
+<% # this template requires safe mode to be disabled -%>
+
+<%= render_template('Service Action - SSH Default', :service => 'puppet', :action => 'stop') -%>
+
+<%- if input('Delete certificate on Puppet CA') == 'yes' %>
+# deleting host certificate on puppet ca proxy
+<%-   if preview? -%>
+#   this would delete a certificate on puppet ca, but running in preview mode
+<%-   else -%>
+<%-     if @host.puppet_ca_proxy.statuses[:puppetca].destroy(@host.name) -%>
+<%=       '# deleted cert on proxy' %>
+<%-     else -%>
+<%=       '# failed to delete the proxy, aborting job' -%>
+<%-       raise('Unable to delete the certificate on Puppet CA proxy, check Foreman proxy logs') -%>
+<%=     end -%>
+<%-   end -%>
+<%- end -%>
+
+# delete the old cert and key
+ssl_path=`puppet config print ssldir`
```

Comment:
Would it make sense to check if it's not empty?

Reply:
wrapped in condition